domain-services/fiscal-event-service/src/main/java/org/egov/validator/FiscalEventValidator.java

```diff
@@ -86,130 +87,87 @@ public void validateFiscalEventPushPost(FiscalEventRequest fiscalEventRequest) {
+        if (StringUtils.isBlank(fiscalEvent.getReferenceId())) {
+            errorMap.put(MasterDataConstants.REFERENCE_ID, "Reference id is missing for event type  : "
+                    + (fiscalEvent.getEventType() != null ? fiscalEvent.getEventType().name() : null));
+            isMissing = true;
+        }
+        if (fiscalEvent.getEventType() == null
+                || !EnumUtils.isValidEnum(FiscalEvent.EventTypeEnum.class, fiscalEvent.getEventType().name())) {
 
-            List<FiscalEvent> fiscalEvents = fiscalEventRequest.getFiscalEvent();
-            Set<String> parentEventIds = new HashSet<>();
-            String tenantId = fiscalEventRequest.getFiscalEvent().get(0).getTenantId();
-            for (FiscalEvent fiscalEvent : fiscalEvents) {
-                if (StringUtils.isNotBlank(fiscalEvent.getLinkedEventId())) {
-                    parentEventIds.add(fiscalEvent.getLinkedEventId());
-                }
-            }
-            if (!parentEventIds.isEmpty() && StringUtils.isNotBlank(tenantId)) {
-                Criteria criteria = new Criteria();
-                List<String> ids = new ArrayList<>(parentEventIds);
-                criteria.setTenantId(tenantId);
-                criteria.setIds(ids);
-
-                List<Object> fiscalEventList = eventRepository.searchFiscalEvent(criteria);
-                if (fiscalEventList == null || fiscalEventList.isEmpty()) {
-                    errorMap.put(MasterDataConstants.PARENT_EVENT_ID, "Linked event id(s) : " + parentEventIds + " doesn't exist in the system.");
-                } else if (!fiscalEventList.isEmpty() && (fiscalEventList.size() != parentEventIds.size())) {
-                    List<FiscalEvent> dbFiscalEvents = mapperUtil.mapDereferencedFiscalEventToFiscalEvent(fiscalEventList);
-                    Set<String> missingParentEventIds = new HashSet<>();
-                    for (String parentEventId : parentEventIds) {
-                        boolean isExist = false;
-                        for (FiscalEvent fiscalEvent : dbFiscalEvents) {
-                            if (fiscalEvent != null && StringUtils.isNotBlank(fiscalEvent.getLinkedEventId())
-                                    && fiscalEvent.getLinkedEventId().equals(parentEventId)) {
-                                isExist = true;
-                                break;
-                            }
-                        }
-                        if (!isExist) {
-                            missingParentEventIds.add(parentEventId);
-                        }
-                    }
-                    if (!missingParentEventIds.isEmpty()) {
-                        errorMap.put(MasterDataConstants.PARENT_EVENT_ID, "Linked event id(s) : " + missingParentEventIds + " doesn't exist in the system.");
-                    }
-                }
-            }
+            errorMap.put(MasterDataConstants.EVENT_TYPE, "Fiscal event type is missing");
+            isMissing = true;
```

Comment:
As `isMissing` is a basic type and not an object, it will be *passByValue* and not *passByReference*. So setting this variable here wouldn't mean anything. It won't reflect back in the calling method's variable's value. 
This should be visible in the IDE.

Reply:
minor change, to be taken up as part of code cleanup.